### ISSUE TITLE
security group change of instance in VPC shouldn't require new resource

### ISSUE THREAD

**Issue by lgtml** [hashicorp/terraform#194]
Security group membership changes on VPC nodes should not be destructive.

To reproduce:
1. Launch a node with default security groups than update the instance. 
2. Run Apply

*note: will open another issue in regards to the security group ordering.*
```
    security_groups.0: "sg-aaaaaaa" => "sg-bbbbbbb" (forces new resource)
    security_groups.1: "sg-bbbbbbb" => "sg-aaaaaaa" (forces new resource)
```

**Comment by davedash** [comment]
I don't think it's being caused by a set issue.  Or rather that issue is solved, this issue remains:

```
    security_groups.#:          "3" => "2" (forces new resource)
    security_groups.3052868674: "sg-e95bb28d" => "sg-e95bb28d" (forces new resource)
    security_groups.3312015480: "sg-a22dc2c6" => "sg-a22dc2c6" (forces new resource)
```

This should just remove the spurious SG since we're in VPC.

**Comment by ggiamarchi** [comment]
@davedash It seems to be the same as https://github.com/hashicorp/terraform/issues/873

**Comment by catsby** [comment]
This may be fixed with #1153 which looks promising but needs to be rebased on master

**Comment by mitchellh** [comment]
It is fixed by #1153 (or at least, the trail starting there)
